core/src/main/java/org/apache/calcite/util/format/postgresql/EnumStringFormatPattern.java

```diff
@@ -31,12 +32,27 @@ public class EnumStringFormatPattern extends StringFormatPattern {
   private final ChronoField chronoField;
   private final String[] enumValues;
 
-  public EnumStringFormatPattern(ChronoField chronoField, String... patterns) {
-    super(patterns);
+  public EnumStringFormatPattern(ChronoUnitEnum chronoUnit, ChronoField chronoField,
+      String... patterns) {
+    super(chronoUnit, patterns);
     this.chronoField = chronoField;
     this.enumValues = patterns;
   }
 
+  @Override protected int parseValue(ParsePosition inputPosition, String input, Locale locale,
+      boolean haveFillMode, boolean enforceLength) throws Exception {
+    final String inputTrimmed = input.substring(inputPosition.getIndex());
+
+    for (int i = 0; i < enumValues.length; i++) {
+      if (inputTrimmed.startsWith(enumValues[i])) {
+        inputPosition.setIndex(inputPosition.getIndex() + enumValues[i].length());
+        return i;
+      }
+    }
+
+    throw new Exception();
```

Comment:
this exception needs a message

Reply:
Changed to a ParseException and added a message.